analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/DeliveringDebugFeaturesInProduction.cs

```diff
@@ -30,9 +30,15 @@ public sealed class DeliveringDebugFeaturesInProduction : DeliveringDebugFeature
         internal /*for testing*/ DeliveringDebugFeaturesInProduction(IAnalyzerConfiguration configuration) : base(configuration) { }
 
         // For simplicity we will avoid creating noise if the validation is invoked in the same method (https://github.com/SonarSource/sonar-dotnet/issues/5032)
-        protected override bool IsDevelopmentCheckInvoked(SyntaxNode node, SemanticModel semanticModel) =>
-            node.FirstAncestorOrSelf<MethodDeclarationSyntax>() is { } parentMethodDeclaration
-            && parentMethodDeclaration.DescendantNodes().Any(x => IsDevelopmentCheck(x, semanticModel));
+        protected override bool IsDevelopmentCheckInvoked(SyntaxNode node, SemanticModel semanticModel)
+        {
+            SyntaxNode parentMethod = node.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+            if (parentMethod is null && node.FirstAncestorOrSelf<GlobalStatementSyntax>() is { }) // for top-level statements
+            {
+                parentMethod = node.FirstAncestorOrSelf<CompilationUnitSyntax>();
+            }
+            return parentMethod is not null && parentMethod.DescendantNodes().Any(x => IsDevelopmentCheck(x, semanticModel));
```

Comment:
This result in a lot of FNs: 
```cs
{
    app.Environment.IsDevelopment();
    app.UseDeveloperExceptionPage();
}
```
Do we have tests for that?

Reply:
[Yes, it's covered.](https://github.com/SonarSource/sonar-dotnet/blob/cff1e6ade43eece40049467405aab5b3307cf91a/analyzers/tests/SonarAnalyzer.Test/TestCases/Hotspots/DeliveringDebugFeaturesInProduction.NetCore2.cs#L123)
There are several FNs. The rule is pretty lenient about this. It doesn't involve symbolic execution or use the control flow in any way. It just checks whether there's a call to the `IsDevelopment()` method in the same scope (regardless of where that call is located).